shared/src/commonMain/kotlin/com/projectnewm/example/ExampleUseCase.kt

```diff
@@ -0,0 +1,29 @@
+package com.projectnewm.example
+
+import com.projectnewm.BaseUseCase
+import com.projectnewm.util.CommonFlow
+import com.projectnewm.util.DataState
+import com.projectnewm.util.asCommonFlow
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.flow.flow
+
+class ExampleUseCase constructor(
+    private val exampleRepository: ExampleRepository
+) : BaseUseCase<List<ExampleDomainModel>> {
+
+    override fun execute(): CommonFlow<DataState<List<ExampleDomainModel>>> =
```

Comment:
Note: The Swift version of this loses type information about the List's generic, so it's just an untyped `NSArray` :/

Reply:
Wondering if this needs to be a flow or can just suspend and return the type?  Then view models would handle loading state and use cases would be synchronous.

Reply:
I think it'll be cleaner to have the use case handle loading/data/error emissions and easier to test since everything is in one flow. Plus it's more code that we can share

Reply:
Yeah, we want to share as much as possible, but I think it's a bad idea to start saying "well, conceptually this *would* go in the view model, but we're not sharing view models, so let's stick it in the use case, purely so we can share it."  If we want to share view models (and the functionality they're intended to contain), we should look more into that.